presto-main/src/main/java/com/facebook/presto/operator/scalar/sql/StringSqlFunctions.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.operator.scalar.sql;
+
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.SqlInvokedScalarFunction;
+import com.facebook.presto.spi.function.SqlParameter;
+import com.facebook.presto.spi.function.SqlParameters;
+import com.facebook.presto.spi.function.SqlType;
+
+public class StringSqlFunctions
+{
+    private StringSqlFunctions() {}
+
+    @SqlInvokedScalarFunction(value = "replace_first", deterministic = true, calledOnNullInput = true)
+    @Description("Replaces the first occurrence of a substring that matches the given pattern with the given replacement.")
+    @SqlParameters({@SqlParameter(name = "str", type = "varchar"), @SqlParameter(name = "search", type = "varchar"), @SqlParameter(name = "replace", type = "varchar")})
+    @SqlType("varchar")
+    public static String replaceFirst()
+    {
+        return "RETURN IF(STRPOS(str, search) = 0, str, CONCAT(SUBSTR(str, 1, STRPOS(str, search) - 1), replace, SUBSTR(str, STRPOS(str, search) + LENGTH(search))))";
+    }
```

Comment:
Try:

IF(CARDINALITY(SPLIT(str, search, 1)) = 2, SPLIT(str, search, 1)[1] || replace || SPLIT(str, search, 1)[2]), str)

strpos is not efficient. Split is more efficient

Reply:
Note that when an expression is repeated, we evaluate it only once.